server/src/utils.ts

```diff
@@ -1,250 +1,24 @@
-        if (maybeModuleXmlData === undefined) {
-          preparedDiagnostic.message = `Could not parse target document!: ${linkTargetModule}`
-          diagnostics.push(preparedDiagnostic)
-          continue
-        }
-        parsedOtherPages.set(linkTargetModule, maybeModuleXmlData)
-      }
-      const targetElements = select(
-        `//cnxml:*[@id="${linkTargetId}"]`,
-        maybeModuleXmlData
-      ) as Node[]
-      if (targetElements.length === 0) {
-        preparedDiagnostic.message = `Target ID in document doesn't exist!: ${linkTargetId}`
-        diagnostics.push(preparedDiagnostic)
-        continue
-      } else if (targetElements.length > 1) {
-        preparedDiagnostic.message = `Target ID in document is not unique!: ${linkTargetId}`
-        diagnostics.push(preparedDiagnostic)
-        continue
-      }
-    }
-  }
-
-  return diagnostics
-}
-
-async function validateSamePageLinks(xmlData: Document): Promise<Diagnostic[]> {
-  const diagnostics: Diagnostic[] = []
-  const select = xpath.useNamespaces({ cnxml: NS_CNXML })
-  const samePageLinks = select('//cnxml:link[@target-id and not(@document)]', xmlData) as Node[]
-  const samePageElementsWithIds = select('//cnxml:*[@id]', xmlData) as Node[]
-  const samePageIdCounts = new Map<string, number>()
-  const validTargetIds = new Set<string>()
-  const duplicateTargetIds = new Set<string>()
 
-  for (const elementWithId of samePageElementsWithIds) {
-    const element = elementWithId as any
-    const elementId = element.getAttribute('id')
-
-    if (elementId === '') {
-      continue
-    }
-
-    const count = samePageIdCounts.get(elementId)
-    if (count === undefined) {
-      samePageIdCounts.set(elementId, 1)
-    } else {
-      samePageIdCounts.set(elementId, count + 1)
-    }
-  }
-
-  samePageIdCounts.forEach((val, key, map) => {
-    if (val === 1) {
-      validTargetIds.add(key)
-    }
-
-    if (val > 1) {
-      duplicateTargetIds.add(key)
-    }
-  })
-
-  for (const link of samePageLinks) {
-    // Check the document for a matching id element
-    const linkElement = link as any
-    const [startPosition, endPosition] = calculateElementPositions(linkElement)
-    const preparedDiagnostic: Diagnostic = generateDiagnostic(
-      DiagnosticSeverity.Error,
-      startPosition,
-      endPosition,
-      '',
-      LINK_DIAGNOSTIC_SOURCE
-    )
-    const linkTargetId: string = linkElement.getAttribute('target-id')
-    if ((linkTargetId === '') || (validTargetIds.has(linkTargetId))) {
-      continue
-    }
-
-    if (duplicateTargetIds.has(linkTargetId)) {
-      preparedDiagnostic.message = `Target for link is not unique!: ${linkTargetId}`
-      diagnostics.push(preparedDiagnostic)
-      continue
-    } else {
-      preparedDiagnostic.message = `Target for link doesn't exist!: ${linkTargetId}`
-      diagnostics.push(preparedDiagnostic)
-      continue
-    }
-  }
-
-  return diagnostics
-}
+const SOURCE = 'cnxml language server'
```

Comment:
Minor / subjective: What do you think about using the shorter `cnxml` value here? Since VScode will end up displaying errors with `${source}(${code})` that may keep things tighter when displayed in the problems view. Other language servers seem to use values like `xml`, `eslint`, etc.